packages/ozone-api/ozone-api-item/src/ozone-api-item.ts

```diff
@@ -5,17 +5,39 @@
 import * as Config from 'ozone-config'
 import {OzoneAPIRequest} from 'ozone-api-request'
 import {customElement, domElement, jsElement} from 'taktik-polymer-typescript'
-import {Item} from 'ozone-type'
+import {Item, ItemSearchResult} from 'ozone-type'
 import {SearchResponse, SearchResult, SearchQuery} from "ozone-search-helper";
+import {v4 as uuid} from 'uuid';
 
-export interface BulkResponse {
-    response:Array<Item>;
+/**
+ * Function decorator decorator to be used to wait until 
+ * all other decorated function resolve.
+ * This decorator is aimed to be used in class that implement StatefulOzone.
+ * It's purpose is to wait others ozone call finish, before sending a next one.
+ * 
+ */
+export function lockRequest() {
+    return function (target: StatefulOzone, propertyKey: string, descriptor: PropertyDescriptor) {
+        let originalMethod = descriptor.value;
+
+        descriptor.value = function() {
+            const self: StatefulOzone = this as any;
+            const arg = arguments;
+
+            self._currentRequest = self._currentRequest
+                .catch()
+                .then(()=> {
+                    return originalMethod.apply(this, arg);
```

Comment:
Here we have this === self since we are in an arrow function. Is it the expected behaviour ?

Reply:
Main reason is to have a type checking with ts compiler